session/session_test.go

```diff
@@ -0,0 +1,176 @@
+package session
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestGetDeveloperPortalDataFromEnv(t *testing.T) {
+	// If the BITRISE_PORTAL_DATA_JSON env is already set (e.g for local testing the step),
+	// then we don't want to modify it during the test
+	var envAlreadyPresented bool
+	if os.Getenv("BITRISE_PORTAL_DATA_JSON") == "" {
+		if err := os.Setenv("BITRISE_PORTAL_DATA_JSON", dummyPortalDataJSON); err != nil {
+			t.Errorf("Failed to set BITRISE_PORTAL_DATA_JSON env to test getDeveloperPortalData() error = %v", err)
+		}
+	} else {
+		envAlreadyPresented = true
+	}
+
+	got, err := getDeveloperPortalData("", "")
+	if err != nil {
+		t.Errorf("getDeveloperPortalData() error = %v", err)
+		return
+	}
+
+	if !envAlreadyPresented {
+		if !reflect.DeepEqual(got, dummyPortalData) {
+			t.Errorf("getDeveloperPortalData() = %v, want %v", sPretty(got), sPretty(dummyPortalData))
+		}
+
+		// Reset the dummy BITRISE_PORTAL_DATA_JSON
+		if err := os.Setenv("BITRISE_PORTAL_DATA_JSON", ""); err != nil {
+			t.Errorf("Failed to reset BITRISE_PORTAL_DATA_JSON env after testing getDeveloperPortalData() error = %v", err)
+		}
+	}
+}
+
+// TestGetDeveloperPortalData
+func TestGetDeveloperPortalData(t *testing.T) {
+	buildURL, builAPIToken := os.Getenv("BITRISE_BUILD_URL"), os.Getenv("BITRISE_BUILD_API_TOKEN")
+	if buildURL == "" {
+		t.Skip("Failed to run TestGetDeveloperPortalData() test because the BITRISE_BUILD_URL env is not exported")
+	}
+	if builAPIToken == "" {
+		t.Skip("Failed to run TestGetDeveloperPortalData() test because the BITRISE_BUILD_API_TOKEN env is not exported")
+	}
+
+	got, err := getDeveloperPortalData(buildURL, builAPIToken)
+	if err != nil {
+		t.Errorf("getDeveloperPortalData() error = %v", err)
+	}
+
+	if reflect.DeepEqual(got, PortalData{}) {
```

Comment:
why is the dev portal data empty?

Reply:
It's testing that the fetched dev portal data is not empty.
There is no option to compare the fetched data to a mock value because this data will be dynamically changed in every month.